### ISSUE TITLE
I can't increase Prometheus startupProbe failureThreshold value

### ISSUE THREAD

**Issue by elmazzun** [openshift/cluster-monitoring-operator#2085]
Hi

Is there a officially supported way to alter Prometheus container startupProbe failureThreshold value?
I need to increase this value because my WAL is getting bigger and bigger.

I tried:

1. altering the ConfigMap `cluster-monitoring-config` did not bring results (because such ConfigMap has no parameter that lets you alter failureThreshold, at least according to https://github.com/openshift/cluster-monitoring-operator/blob/master/pkg/manifests/types.go#L36)
2. altering the Prometheus instance as suggested in https://github.com/prometheus-operator/prometheus-operator/issues/4730 and https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/user-guides/strategic-merge-patch.md with the following command:
```bash
$ oc patch prometheus/k8s -n openshift-monitoring --type='merge' -p "$(cat <<- EOF
spec:
  containers:
  - name: prometheus
    startupProbe:
      failureThreshold: 500
EOF
)"
```
This modification was applied to Prometheus StatefulSet too and a redeploy was triggered but after 10 minutes the Prometheus `k8s`instance was reverted to default values.

Is there a way to increase probes parameters?
My cluster is getting pretty big and I need Prometheus to scale its probes too accorging to my cluster.

My environment:
* OCP 4.11
* Prometheus 2.36.2

**Comment by simonpasquier** [comment]
The startup probe delay should be 1h already in 4.11. Isn't it sufficient in your case?
There's no workaround except moving CMO to unmanaged state.

**Comment by elmazzun** [comment]
> The startup probe delay should be 1h already in 4.11. Isn't it sufficient in your case?

At the moment, Prometheus has the following probe:
```yaml
startupProbe:
  exec:
    command:
      - sh
      - '-c'
      - >-
        if [ -x "$(command -v curl)" ]; then exec curl --fail
        http://localhost:9090/-/ready; elif [ -x "$(command -v wget)"
        ]; then exec wget -q -O /dev/null
        http://localhost:9090/-/ready; else exit 1; fi
  timeoutSeconds: 3
  periodSeconds: 15
  successThreshold: 1
  failureThreshold: 60
```

If I am right, Prometheus will have a maximum of 60 * 15 seconds to complete its startup.

> There's no workaround except moving CMO to unmanaged state.

Would that let me modify startupProbe parameters?

**Comment by simonpasquier** [comment]
the startup probe timeout has been increased to 60m in 0bfc262fba06e6660f5ab5180a5def35e852eaf7 which is included in OCP 4.11.21 (https://access.redhat.com/errata/RHSA-2022:9107). If you're on OKD, I'm not sure about the corresponding version.

**Comment by elmazzun** [comment]
My Cluster OCP **4.11.12**, that is why my parameters are so low.

Guess I'll have to give up to update probes parameters and just update my 4.11.12 Cluster to at least 4.11.21.

Thank you very much @simonpasquier
